ros2doctor/ros2doctor/api/package.py

```diff
@@ -0,0 +1,156 @@
+    """
+    Return warning messages for PackageCheck, and info for PackageReport.
+
+    :param: dictionary of local package name and version
+    :param: dictionary of rosdistro package name and version
+    :param: boolean value determines which output to populate, msgs or report
+    :return: list of warning messages
+    """
+    warning_msgs = []
+    missing_req = ''
+    missing_local = ''
+    for name, local_ver_str in local_packages.items():
+        if not local_ver_str:
+            missing_local += ' ' + name
+            local_ver_str = ''
+        required_ver_str = distro_packages.get(name)
+        if not required_ver_str:
+            missing_req += ' ' + name
+            required_ver_str = ''
+        local_ver = version.parse(local_ver_str).base_version
+        required_ver = version.parse(required_ver_str).base_version
+        if local_ver < required_ver:
+            warning_msgs.append(f'{name} has been updated to a new version.'
+                                f' local: {local_ver} <'
+                                f' required: {required_ver}')
+    if missing_req:
+        warning_msgs.append('Cannot find required versions of packages:' + missing_req)
+    if missing_local:
+        warning_msgs.append('Cannot find local versions of packages:' + missing_local)
+    return warning_msgs
+
+
+class PackageCheck(DoctorCheck):
+    """Check local package versions against release versions on rosdistro."""
+
+    def category(self):
+        return 'package'
+
+    def check(self):
+        """Check packages within the directory where command is called."""
+        result = Result()
+        try:
+            distro_package_vers = get_distro_package_versions()
+            if not distro_package_vers:
+                result.add_error('ERROR: distro packages info is not found.')
+        except (AttributeError, RuntimeError, URLError):
+            result.add_error('ERROR: Unable to fetch package information from rosdistro.')
+        local_package_vers = get_local_package_versions()
+        if not local_package_vers:
+            result.add_error('ERROR: local package info is not found.')
+        if result.error != 0:
+            return result
+
+        warning_msgs = compare_versions(local_package_vers, distro_package_vers)
+        for msg in warning_msgs:
+            result.add_warning(msg)
+        return result
+
+
+class PackageReport(DoctorReport):
+    """Report local package versions and release versions on rosdistro."""
+
+    def category(self):
+        return 'package'
+
+    def report(self):
+        """Report packages within the directory where command is called."""
+        report = Report('PACKAGE VERSIONS')
+        try:
+            distro_package_vers = get_distro_package_versions()
+        except (AttributeError, RuntimeError, URLError):
+            return report
+        local_package_vers = get_local_package_versions()
+
+        if local_package_vers and distro_package_vers:
```

Comment:
Suggestion: you can return early if this is not true and avoid additional indentation.

```python
if not local_package_vers or not distro_package_vers:
    return report
```